src/Mailgun/Api/Message.php

```diff
@@ -20,59 +20,74 @@
 class Message extends HttpApi
 {
     /**
-     * @param $domain
-     * @param array $params
+     * @param string $domain
+     * @param array  $params
      *
      * @return SendResponse
      */
     public function send($domain, array $params)
     {
+        Assert::string($domain);
         Assert::notEmpty($domain);
         Assert::notEmpty($params);
 
         $postDataMultipart = [];
-        $fields = ['message', 'attachment', 'inline'];
+        $fields = ['attachment', 'inline'];
         foreach ($fields as $fieldName) {
             if (!isset($params[$fieldName])) {
                 continue;
             }
-            if (!is_array($params[$fieldName])) {
-                $postDataMultipart[] = $this->prepareFile($fieldName, $params[$fieldName]);
-            } else {
-                foreach ($params[$fieldName] as $file) {
-                    $postDataMultipart[] = $this->prepareFile($fieldName, $file);
-                }
+
+            Assert::isArray($params[$fieldName]);
+            foreach ($params[$fieldName] as $file) {
+                $postDataMultipart[] = $this->prepareFile($fieldName, $file);
             }
 
             unset($params[$fieldName]);
         }
 
-        foreach ($params as $key => $value) {
-            if (is_array($value)) {
-                foreach ($value as $subValue) {
-                    $postDataMultipart[] = [
-                        'name' => $key,
-                        'content' => $subValue,
-                    ];
-                }
-            } else {
-                $postDataMultipart[] = [
-                    'name' => $key,
-                    'content' => $value,
-                ];
-            }
-        }
-
+        $postDataMultipart = array_merge($this->prepareMultipartParameters($params), $postDataMultipart);
         $response = $this->httpPostRaw(sprintf('/v3/%s/messages', $domain), $postDataMultipart);
 
         return $this->hydrateResponse($response, SendResponse::class);
     }
 
     /**
+     * @param string $domain
+     * @param array  $to      with all you send emails to. Including bcc and cc
```

Comment:
I think the `@param` comment here is confusing, specially because `@param` name is `$to` and comment says `including BCC and CC`. If this parameter expects **all** recipients then I suggest to rename the parameter name